### PR TITLE
add DataKinds to the checkExtensions list

### LINKED ISSUE #201: TH should give error about missing `DataKinds` extension
Currently, use of `makeSem`(`_`) without `DataKinds` results in error:
```
Parsemy.hs:18:1-16: error:
    Illegal type: ‘"Parsemy"’ Perhaps you intended to use DataKinds
   |
18 | makeSem ''Parser
   | ^^^^^^^^^^^^^^^^
```
which is not bad, but does not explicitly say that we depend on that extension.
The only change needed should be adding extension to list of required ones [here](https://github.com/polysemy-research/polysemy/blob/829e3202fc7d09faaf757cddd11a8a741b687a70/src/Polysemy/Internal/TH/Effect.hs#L125).
https://github.com/polysemy-research/polysemy/blob/829e3202fc7d09faaf757cddd11a8a741b687a70/src/Polysemy/Internal/TH/Effect.hs#L125

### PR DESCRIPTION
**Pull request by pepegar**
fixes #201

### CODE REVIEW

File: src/Polysemy/Internal/TH/Effect.hs [hunk 1]
```diff
@@ -171,4 +171,3 @@ genDec should_mk_sigs cli = do
                  []
         ]
     ]
-
```
Comment (isovector) [inline src/Polysemy/Internal/TH/Effect.hs]: nit: please revert this change

Comment (pepegar) [reply src/Polysemy/Internal/TH/Effect.hs]: You're right, sorry :)

**Comment by isovector** [comment]
thanks!
